src/devices/Vcnl4040/Internal/AlsConfRegister.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Device.I2c;
+using Iot.Device.Vcnl4040.Definitions;
+
+namespace Iot.Device.Vcnl4040.Internal
+{
+    /// <summary>
+    /// ALS configuration register
+    /// Command code / address: 0x00 (LSB)
+    /// Documentation: datasheet (Rev. 1.7, 04-Nov-2020 9 Document Number: 84274).
+    /// </summary>
+    internal class AlsConfRegister : Register
+    {
+        private const byte AlsItMask = 0b1100_0000;
+        private const byte AlsPersMask = 0b0000_1100;
+        private const byte AlsIntEnMask = 0b0000_0010;
+        private const byte AlsSdMask = 0b0000_0001;
+        private bool _alsItChanged = false;
+        private bool _alsPersChanged = false;
+        private bool _alsIntEnChanged = false;
+        private bool _alsSdChanged = false;
+        private PowerState _alsSd = PowerState.PowerOff;
+        private AlsInterrupt _alsIntEn = AlsInterrupt.Disabled;
+        private AlsInterruptPersistence _alsPers = AlsInterruptPersistence.Persistence1;
+        private AlsIntegrationTime _alsIt = AlsIntegrationTime.Time80ms;
+
+        /// <summary>
+        /// Gets or sets the ALS integration time setting (ALS_IT).
+        /// </summary>
+        public AlsIntegrationTime AlsIt
```

Comment:
We always prefer a proper naming for public properties (even for internal classes). You can call is AlsIntegrationTime, same for the following ones, use directly the name of the enum, at least it's a bit easier to follow and understand what it means.

Reply:
I would like to ask you to reconsider this. Here's why:

The register classes represent the low-level internals of the device. They encapsulate technical details. Regular users of the binding will never come into contact with these details, as all functions are available through a high-level API. That's why the register classes are marked as "internal." For 99.9% of all users, these classes are irrelevant.

However, if someone does use, extend, or debug the binding at the register level, in my opinion, it is much better for understanding if a direct connection can be made between the code and the datasheet. Personally, I don't want to take the detour through a datasheet but rather see it directly in the code (debugger). This seems much clearer to me than a (supposedly) descriptive name for a register or bit group.

Therefore I really would like to stick with the actual names of the things.

Reply:
If those are the names in the datasheet, as it's internal, I'm fine.